### PR TITLE
Building in docker

### PR DESCRIPTION
**Pull request by wdwatkins**
This adds the docker source code, a few little fixes to make the build go faster, and the jenkinsfile

### CODE REVIEW

File: scripts/fetch/dv_data.R [hunk 1]
```diff
@@ -12,7 +12,7 @@ fetch.dv_data <- function(viz){
   sites <- deps[["sites"]]
   
   startDate <- paste0(year, "-01-01")
-  endDate <- paste0(year, "-12-31")
+  endDate <- paste0(year, "-01-02")
```
Comment (aappling-usgs) [inline scripts/fetch/dv_data.R]: Safe to revert this change now?

File: scripts/fetch/dv_sites.R [hunk 2]
```diff
@@ -12,7 +12,7 @@ fetch.dv_sites <- function(viz){
   year <- deps[["year"]][["year"]]
   
   startDate <- paste0(year, "-01-01")
-  endDate <- paste0(year, "-12-31")
+  endDate <- paste0(year, "-01-02")
```
Comment (aappling-usgs) [inline scripts/fetch/dv_sites.R]: revert this change?

File: viz.yaml [hunk 3]
```diff
@@ -18,7 +18,7 @@ info:
       version: 0.9-7
     dataRetrieval:
       repo: CRAN
-      version: 2.7.4
+      version: 2.7.3
```
Comment (aappling-usgs) [inline viz.yaml]: does the older version work better?

File: viz.yaml [hunk 4]
```diff
@@ -28,6 +28,12 @@ info:
     sp:
       repo: CRAN
       version: 1.2.7
+    sbtools:
+      repo: CRAN
+      version: 1.1.6
+    geojsonio:
+      repo: CRAN
+      version: 0.6.0
```
Comment (aappling-usgs) [inline viz.yaml]: It'd be nice to think about removing all of these declared package dependencies as we switch to Docker (because now the Dockerfile is where the decisions actually get made about which package version gets used). Is now too soon to do so, and if not, is it as simple as deleting all the lines in this subsection, or does vizlab give errors if you do that?

File: scripts/fetch/dv_data.R (hunk 1 above)
Comment (wdwatkins) [reply scripts/fetch/dv_data.R]: Yes, although really I'd say these dates should all be moved into the viz.yaml so they can be set in a single place

File: viz.yaml (hunk 3 above)
Comment (wdwatkins) [reply viz.yaml]: It was just to avoid having to set package installs to GRAN, since CRAN only has 2.7.3

File: scripts/fetch/dv_data.R (hunk 1 above)
Comment (aappling-usgs) [reply scripts/fetch/dv_data.R]: Fair. Up to you how you do it, but note that we're not currently sure that anybody will be developing this interactive viz any more this year, so code quality/maintainability matters somewhat less than usual.

File: viz.yaml (hunk 3 above)
Comment (aappling-usgs) [reply viz.yaml]: oh, OK

File: scripts/fetch/dv_data.R (hunk 1 above)
Comment (wdwatkins) [reply scripts/fetch/dv_data.R]: Right, probably wouldn't bother doing it now

File: viz.yaml (hunk 4 above)
Comment (wdwatkins) [reply viz.yaml]: We just would need to remove the `checkVizPackages()` call in`vizmake()`, since that requires there to be a packages yaml block.  Otherwise, I think we could probably just drop the packages section, unless we don't feel confident relying on the package versions that come through the rocker images.   

There is probably a fair bit of code associated with `checkVizPackages` that could either be removed from `vizlab`, unless we wanted to preserve a way of checking package versions against some reference.

Comment (aappling-usgs) [reply viz.yaml]: OK, thanks for thinking it through. I'll create a GitHub issue in vizlab for this, and we can resolve it (including cleaning up the extra code) after we've had a chance to kick the tires on the Docker option for a while.

What happens if you make a packages YAML block but leave it empty?

Comment (aappling-usgs) [reply viz.yaml]: Added this issue: https://github.com/USGS-VIZLAB/vizlab/issues/395

Comment (wdwatkins) [reply viz.yaml]: That just returns a null from `checkVizPackages`, I think that would work
